hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -688,6 +689,7 @@ def greedy_shrink(self):
         """
         self.fixate_shrink_passes(
             [
+                "zero_examples",
```

Comment:
should this be the first pass after the node programs instead? To avoid replacing an example with trivial components only to immediately delete it.

Reply:
I think it seems plausible that either way would be better. There are are actually a lot of cases where `zero_examples` deletes more than `node_programs` would (e.g it can collapse an entire list to empty, or an entire subtree of a recursive strategy to a trivial leaf) because it can remove high level structure. It also makes the test case more redundant, so deletions from `node_programs` may cache more, but I don't know if that actually matters.

Reply:
that makes sense to me! I'm happy to leave this first in the absence of any empirical evidence.

Reply:
hang on, don't we have a shrinker benchmarking setup lying around from all the recent IR work?  I'd love to measure each option and decide that way if we can!